tests/async.rs

```diff
@@ -356,3 +356,33 @@ fn shutdown() -> io::Result<()> {
         Ok(())
     })
 }
+
+// prevent source from unregistering by trying to register it twice
+#[test]
+fn duplicate_socket_insert() -> io::Result<()> {
+    future::block_on(async {
+        let listener = Async::<TcpListener>::bind(([127, 0, 0, 1], 0))?;
+        let addr = listener.as_ref().local_addr()?;
+
+        // attempt to register twice
+        assert!(
+            Async::<TcpListener>::bind(addr).is_err(),
```

Comment:
```suggestion
            Async::<TcpListener>::new(&listener).is_err(),
```

Reply:
@notgull only additional change to suggestion is removing the type annotation